kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/core/v1/ConfigMapOperationsImpl.java

```diff
@@ -0,0 +1,84 @@
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.dsl.internal.core.v1;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.ConfigMapList;
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.fabric8.kubernetes.client.Client;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.dsl.FromFileCreatableOperation;
+import io.fabric8.kubernetes.client.dsl.FromFileCreatableResource;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.fabric8.kubernetes.client.dsl.internal.HasMetadataOperation;
+import io.fabric8.kubernetes.client.dsl.internal.HasMetadataOperationsImpl;
+import io.fabric8.kubernetes.client.dsl.internal.OperationContext;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+import static io.fabric8.kubernetes.client.utils.KubernetesResourceUtil.createNewConfigMapWithEntry;
+import static io.fabric8.kubernetes.client.utils.KubernetesResourceUtil.mergeConfigMapData;
+
+public class ConfigMapOperationsImpl extends HasMetadataOperation<ConfigMap, ConfigMapList, Resource<ConfigMap>>
+    implements FromFileCreatableResource<ConfigMap>, FromFileCreatableOperation<ConfigMap, ConfigMapList, Resource<ConfigMap>> {
+
+  public ConfigMapOperationsImpl(Client client) {
+    this(HasMetadataOperationsImpl.defaultContext(client));
+  }
+
+  public ConfigMapOperationsImpl(OperationContext context) {
+    super(context.withPlural("configmaps"), ConfigMap.class, ConfigMapList.class);
+  }
+
+  @Override
+  public ConfigMapOperationsImpl newInstance(OperationContext context) {
+    return new ConfigMapOperationsImpl(context);
+  }
+
+  @Override
+  public FromFileCreatableResource<ConfigMap> fromFile(String key, Path dirOrFilePath) {
+    return new ConfigMapOperationsImpl(context.withItem(createNewOrMergeWithExistingConfigMap(key, dirOrFilePath)));
+  }
+
+  @Override
+  public ConfigMapOperationsImpl withName(String name) {
+    ConfigMap item = getItem();
+    if (item != null) {
+      if (item.getMetadata() != null) {
+        item.getMetadata().setName(name);
+      } else {
+        item.setMetadata(new ObjectMetaBuilder().withName(name).build());
+      }
+    }
+    return new ConfigMapOperationsImpl(context.withName(name).withItem(item));
+  }
+
+  private ConfigMap createNewOrMergeWithExistingConfigMap(String key, Path dirOrFilePath) {
+    try {
+      ConfigMap item = getItem();
+      ConfigMap tempConfigMap = createNewConfigMapWithEntry(key, dirOrFilePath);
+
+      ConfigMap mergedConfigMap = mergeConfigMapData(item, tempConfigMap);
```

Comment:
Couldn't this be done with just the builder logic? https://github.com/fabric8io/kubernetes-client/compare/master...shawkins:rohanKanojia-pr/issue4184?expand=1#diff-7ff2d3334a0db4e97ce2cd61970f03169cb05b1ee85885b1fcddd79230bf9196R59 

Or is the merging logic needed for another purpose?